core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -801,6 +838,8 @@ private class BufferReleasingInputStream(
     if (!closed) {
       delegate.close()
       iterator.releaseCurrentResultBuffer()
+      // Unset the flag when a remote request finished.
+      if (isNetworkReqDone) NettyUtils.isNettyOOMOnShuffle = false
       closed = true
```

Comment:
Move this intro try/finally ? close exceptions can get ignored; while `isNettyOOMOnShuffle` will never get unset